source/extensions/clusters/dynamic_forward_proxy/cluster.cc

```diff
@@ -311,12 +311,10 @@ Cluster::LoadBalancer::chooseHost(Upstream::LoadBalancerContext* context) {
   }
 
   const Router::StringAccessor* dynamic_host_filter_state = nullptr;
-  if (context->downstreamConnection()) {
+  if (context->requestStreamInfo()) {
```

Comment:
The `requestStreamInfo()` can never return nullptr, since `Network::Connection::streamInfo()` returns a reference. It makes sense to return the reference from the `requestStreamInfo` and not deal with nullptr check.

Reply:
It will return ``nullptr`` for ``LoadBalancerContext`` implementors which didn't override this method. If you take ``TcpProxy`` for instance, it extends ``LoadBalancerContextBase``, which by default will return ``nullptr``:

https://github.com/envoyproxy/envoy/blob/accc5716727bb34c050100f33a25bc4ae0cba31d/source/common/upstream/load_balancer_impl.h#L225

So part of this PR was to override that method and return the pointer:
https://github.com/envoyproxy/envoy/pull/29539/files#diff-6d626cbe0adca12f22f71c49d45cbbdbf250cb2ac648e1117d0cdabb648b33dc